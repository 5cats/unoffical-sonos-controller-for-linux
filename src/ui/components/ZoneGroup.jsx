import register from '../helpers/registerComponent';
import model from '../model';
import ZoneGroupMember from './ZoneGroupMember'; 

class ZoneGroup {

	componentDidMount () {
		var self = this;

		model.observe('currentZone', function () {
			self.forceUpdate();
		});
	}

	componentWillUnmount () {
		// unsubscribe
	}

	render () {

		var zoneNodes = this.props.data.ZoneGroupMember.map(function (z) {
			return (
				<ZoneGroupMember data={z.$} />
			);
		});

		var classString = 'not-selected'

		if(model.currentZone.$.ID === this.props.data.$.ID) {
			classString = 'selected';
		}

			return (
				<ul className={classString} onClick={this._onClick}>
					{{zoneNodes}}
				</ul>
		);
	}

	_onClick () {
		model.currentZone = this.props.data;
	}
}

ZoneGroup.prototype.displayName = "ZoneGroup";
export default register(ZoneGroup);