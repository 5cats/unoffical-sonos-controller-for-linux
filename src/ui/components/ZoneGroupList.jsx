import { h } from 'preact';
import { connect } from 'preact-redux';

import {
    getZoneGroups,
    getCurrentHost,
    getPlayStates
} from '../selectors/ZoneGroupSelectors';

import ZoneGroup from './ZoneGroup';

const mapStateToProps = state => {
    return {
        groups: getZoneGroups(state),
        playStates: getPlayStates(state),
        currentHost: getCurrentHost(state)
    };
};

export function ZoneGroupList(props) {
    const zoneGroupNodes = Object.keys(props.groups || {}).map(key => {
        const item = props.groups[key];
        return <ZoneGroup key={key} group={item} {...props} />;
    });

    return (
        <div
            id="zone-container-inner"
            style="width:100%;display: flex; flex-direction: column;"
        >
            <div style="overflow-y: auto;">
                <div id="zone-wrapper">
                    {zoneGroupNodes}
                </div>
            </div>
        </div>
    );
}

export default connect(mapStateToProps, null)(ZoneGroupList);
