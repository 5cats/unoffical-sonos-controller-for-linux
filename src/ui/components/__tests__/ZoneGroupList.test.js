import React from 'react';
import { ZoneGroupList } from '../ZoneGroupList';
import ZoneGroup from '../ZoneGroup';
import { deep } from 'react-render-spy';

jest.mock('../ZoneGroup');
ZoneGroup.mockReturnValue(<p />);

describe('ZoneGroupList', () => {
    let props;

    beforeEach(() => {
        props = {};
    });

    it('matches snapshot', () => {
        const context = deep(<ZoneGroupList {...props} />);
        expect(context.output()).toMatchSnapshot();
    });

    it('renders wrapper divs', () => {
        const context = deep(<ZoneGroupList {...props} />);
        expect(context.find('#zone-container-inner').length).toBe(1);
        expect(context.find('#zone-wrapper').length).toBe(1);
    });

    it('renders group items', () => {
        props.groups = {
            1: {
                foo: true
            },
            2: {
                foo: false
            }
        };

        const context = deep(<ZoneGroupList {...props} />);
        expect(context.output()).toMatchSnapshot();

        expect(ZoneGroup).toHaveBeenCalledTimes(2);

        const [firstCall, lastCall] = ZoneGroup.mock.calls;

        expect(firstCall[0]).toMatchObject({
            group: { foo: true },
            groups: props.groups
        });

        expect(lastCall[0]).toMatchObject({
            group: { foo: false },
            groups: props.groups
        });
    });
});
