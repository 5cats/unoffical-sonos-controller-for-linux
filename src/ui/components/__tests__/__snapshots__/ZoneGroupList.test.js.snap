// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneGroupList matches snapshot 1`] = `
<div
  id="zone-container-inner"
  style="width:100%;display:flex;flex-direction:column"
>
  <div
    style="overflow-y:auto"
  >
    <div
      id="zone-wrapper"
    />
  </div>
</div>
`;

exports[`ZoneGroupList renders group items 1`] = `
<ZoneGroupList
  groups={
    Object {
      "1": Object {
        "foo": true,
      },
      "2": Object {
        "foo": false,
      },
    }
  }
>
  <div
    id="zone-container-inner"
    style={
      Object {
        "display": "flex",
        "flexDirection": " column",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "overflowY": "auto",
        }
      }
    >
      <div
        id="zone-wrapper"
      >
        <Component
          group={
            Object {
              "foo": true,
            }
          }
          groups={
            Object {
              "1": Object {
                "foo": true,
              },
              "2": Object {
                "foo": false,
              },
            }
          }
          key="1"
        >
          <p />
        </Component>
        <Component
          group={
            Object {
              "foo": false,
            }
          }
          groups={
            Object {
              "1": Object {
                "foo": true,
              },
              "2": Object {
                "foo": false,
              },
            }
          }
          key="2"
        >
          <p />
        </Component>
      </div>
    </div>
  </div>
</ZoneGroupList>
`;
